domain/application/state/types.go

```diff
@@ -680,3 +681,36 @@ type containerImageMetadata struct {
 	UserName     string `db:"username"`
 	Password     string `db:"password"`
 }
+
+type setResource struct {
+	// UUID is resources unique identifier.
+	UUID string `db:"uuid"`
+	// CharmUUD is the UUID of the charm this resource is defined in.
+	CharmUUID string `db:"charm_uuid"`
+	// CharmResourceName is the name of the resource in the charm definition.
+	CharmResourceName string `db:"charm_resource_name"`
+	// Revision is the resource revision.
+	Revision sql.Null[int64] `db:"revision"`
+	// OriginTypeID is the ID of the origin of the stored resource.
+	OriginTypeID int `db:"origin_type_id"`
+	// Created at is the time the row is inserted into the resource table.
+	CreatedAt time.Time `db:"created_at"`
+	// StateID is the state ID of the resource
```

Comment:
Question: What is StateID?

Reply:
Good question :) I found it very unclear when writing this PR, I left in the TODO elsewhere in the PR to ask you when you got back. I've chatted with Guillaume who explained it to me today. I will hold off updating the PR w.r.t this until we've had a chat.

Reply:
FYK i added a state enum in https://github.com/juju/juju/pull/18460

Reply:
right, bad memory